src/main/scala/is/hail/sparkextras/OrderedRDD.scala

```diff
@@ -148,26 +142,28 @@ object OrderedRDD {
         info.min == info.max && info.min == max
       }
 
-      while (it.hasNext && overlaps(sortedKeyInfo(it.head)))
+      while (it.hasNext && overlaps(sortedKeyInfo(it.head))) {
+        anyOverlaps = true
         indicesBuilder += it.next()
+      }
 
-      if (it.hasNext && sortedKeyInfo(it.head).min == max)
+      if (it.hasNext && sortedKeyInfo(it.head).min == max) {
```

Comment:
I think you can nuke `overlaps` and the previous loop and make this loop `while (it.hasNext && ....min == max)`.